### PR TITLE
refactor(console): keep sign-in method verification code check state

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

Unexpected behavior: the verification code should not be automatically checked when switching from one identifier to another.

[Video](https://user-images.githubusercontent.com/10806653/215941145-5f1dc57f-84cf-434f-bfe0-b1b400161415.mov)

## Testing

Feb-01-2023 11-53-32

### CODE REVIEW

File: packages/console/src/pages/SignInExperience/tabs/SignUpAndSignIn/components/SignInMethodEditBox/utilities.ts [hunk 1]
```diff
@@ -12,6 +15,28 @@ export const getSignInMethodPasswordCheckState = (
...
-export const getSignInMethodVerificationCodeCheckState = (signInIdentifier: SignInIdentifier) => {
-  return signInIdentifier !== SignInIdentifier.Username;
+export const getSignInMethodVerificationCodeCheckState = (
+  signInIdentifier: SignInIdentifier,
+  signUpConfig: SignUpForm,
+  currentCheckState?: boolean
+) => {
+  if (signInIdentifier === SignInIdentifier.Username) {
+    return false;
+  }
```
Comment (simeng-li) [inline packages/console/src/pages/SignInExperience/tabs/SignUpAndSignIn/components/SignInMethodEditBox/utilities.ts:25]: should return the currentCheckState SSOT?

File: packages/console/src/pages/SignInExperience/tabs/SignUpAndSignIn/SignUpForm.tsx [hunk 2]
```diff
@@ -80,7 +81,11 @@ const SignUpForm = () => {
         {
           identifier: requiredIdentifier,
           password: getSignInMethodPasswordCheckState(requiredIdentifier, isSignUpPasswordRequired),
-          verificationCode: getSignInMethodVerificationCodeCheckState(requiredIdentifier),
+          verificationCode: getSignInMethodVerificationCodeCheckState(
```
Comment (simeng-li) [inline packages/console/src/pages/SignInExperience/tabs/SignUpAndSignIn/SignUpForm.tsx:84]: According to the issue description, should not update the verification code value in `signInMethods`?   Why do we need to evaluate the `verificationCode` here?

**Review by simeng-li — CHANGES_REQUESTED** [review]

File: packages/console/src/pages/SignInExperience/tabs/SignUpAndSignIn/components/SignInMethodEditBox/utilities.ts (hunk 1 above)
Comment (xiaoyijun) [reply packages/console/src/pages/SignInExperience/tabs/SignUpAndSignIn/components/SignInMethodEditBox/utilities.ts:25]: No, If the sign-in method is Username, the verification code state is set to false by default.

**Review by simeng-li — APPROVED** [review]
LGTM

**Pull request merged**